modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,110 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	/// has_taur_sprite[icon_state]: Does this icon_state have ANY taur sprite?
+	var/static/list/has_taur_sprite = list()
+
+	/// has_taur_sprite_suit[icon_state]: has_taur_sprite but for /obj/item/clothng/suit
+	var/static/list/has_taur_sprite_suit = list()
+
+/obj/item/clothing/Initialize(mapload)
+	. = ..()
+	handle_taur_sprites()
+
+/* NOTE FOR handle_taur_sprites(): This can be converted to handle things like digi sprites, vox, etc. I just didn't do it because I don't want to refactor the entire
+system for this one pr. ~niko */
+
+/**
+ * Called in /obj/item/clothing/Initialize().
+ *
+ * Quits instantly if mutant_variants has all taur styles (STYLE_TAUR), then checks to see if the clothing's icon state has an alt sprite for taurs. If no, it then checks to see if
+ * the icon state EXPLICITELY has no taur sprites. If no, it then checks 3 DMI files for the icon_state, and if it appears in any of them, has_taur_sprite is set to
+ * true for that icon state, then it sets the more specific list to true for that state, and finally the specific mutant_variants bitflag for the clothing.
+ *
+ * Finally, if all of the conditions are false, has_taur_sprite for that icon state will be set to false.
+ *
+ * Has seperate logic for suits, as they use a different set of DMIs as of now.
+ */
+/obj/item/clothing/proc/handle_taur_sprites()
+
+	if (istype(src, /obj/item/clothing/suit)) //override for suits, which use different dmis
+
+		if (mutant_variants & STYLE_TAUR)
+			return
+
+		if (has_taur_sprite_suit[icon_state]) // Very slightly speeds up the proc by letting things with no sprites instantly return
+			if (GLOB.naga_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_SNAKE
+			if (GLOB.horse_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_HOOF
+			if (GLOB.pawed_taur_suit_sprites[icon_state])
+				mutant_variants |= STYLE_TAUR_PAW
+			return
+
+		else if (has_taur_sprite_suit[icon_state] == FALSE)
```

Comment:
don't need this, can just be an else

Reply:
eh, sure

Reply:
this causes a possibly unreachable code error

Reply:
how? the caching?

Reply:
Yep, DM thinks it can never reach that if it's an else.

Reply:
Specifically, the error appears on var/taur_sprite = FALSE